benches/benches/bevy_ecs/change_detection.rs

```diff
@@ -243,3 +265,109 @@ fn none_changed_detection(criterion: &mut Criterion) {
         );
     }
 }
+fn insert_if_bit_enabled<const B: u16>(entity: &mut EntityWorldMut, i: u16) {
+    if i & 1 << B != 0 {
+        entity.insert(Data::<B>(1.0));
+    }
+}
+fn add_archetypes_entites<T: Component + Default>(
+    world: &mut World,
+    archetype_count: u16,
+    entity_count: u32,
+) {
+    for i in 0..archetype_count {
+        for _j in 0..entity_count {
+            let mut e = world.spawn(T::default());
+            insert_if_bit_enabled::<0>(&mut e, i);
```

Comment:
So only the first 16 archetypes will have entities, and the other archetypes will be empty?
It might be worth adding a comment

Reply:
It should support at most (2^16)-1 archetypes.
![image](https://github.com/bevyengine/bevy/assets/45868716/cd27b13b-5518-4e15-8611-6a783cc5fdd2)